pkg/lib/oauth/oidc/id_token.go

```diff
@@ -281,30 +260,31 @@ func (ti *IDTokenIssuer) GetUserInfo(userID string, clientLike *oauth.ClientLike
 	out[string(model.ClaimUserCanReauthenticate)] = user.CanReauthenticate
 	out[string(model.ClaimAuthgearRoles)] = roleKeys
 
-	nonPIIUserClaimsOnly := true
-	// When the client is first party
-	// always include userinfo for the userinfo endpoint
-	// We check the scopes only for third party client
 	if clientLike.IsFirstParty {
-		nonPIIUserClaimsOnly = false
+		// When the client is first party, we always include all standard attributes, all custom attributes.
+		for k, v := range user.StandardAttributes {
+			out[k] = v
+		}
+
+		out["custom_attributes"] = user.CustomAttributes
+		out["x_web3"] = user.Web3
 	} else {
-		for _, s := range UserinfoScopes {
-			if slice.ContainsString(clientLike.Scopes, s) {
-				nonPIIUserClaimsOnly = false
+		// When the client is third party, we include the standard claims according to scopes.
+		for k, v := range user.StandardAttributes {
+			isAllowed := false
+			for _, scope := range clientLike.Scopes {
+				if ScopeAllowsClaim(scope, k) {
+					isAllowed = true
+					break
+				}
+			}
+
+			if isAllowed {
+				out[k] = v
 			}
 		}
```

Comment:
What about custom attributes in third party app?

Reply:
Currently custom attributes are not visible in third party app. I think we need a new feature to make them visible.

The developer should configure which third party application can request the scope to see a custom attribute.